### ISSUE TITLE
Type casting not necessary ignores mysqli_fetch_array return type of array|null

### ISSUE THREAD

**Issue by laurin1** [kalessil/phpinspectionsea#1579]
Just started when I updated PHPStorm to 2020.2 Build 202.4357.31. Inspection is flagging this pattern as type casting not necessary (for the array):

```php
		$aReturn = [];

		if($rSQL)
			$aReturn =
				mysqli_fetch_array(
					$rSQL);

		return (array) $aReturn;
```

![image](https://user-images.githubusercontent.com/75857/83408950-1a23b580-a3d9-11ea-9f25-9319c29329c8.png)

![image](https://user-images.githubusercontent.com/75857/83408959-1ee86980-a3d9-11ea-8109-1e0c7de2135b.png)

**Comment by ea-inspections-team** [comment]
Thanks for reporting @laurin1 ! Looks like this is related to PhpStorm stubs, which they've greatly improved. Closing the issue as we are relying to types resolving from IDE here.
